polymer/src/amp-my-embed.html

```diff
@@ -0,0 +1,182 @@
+        @apply --layout-center-justified;
+        text-align: center;
+        margin: 0 48px;
+      }
+
+      .title {
+        color: var(--app-primary-color);
+        font-weight: bold;
+      }
+
+      .description {
+        color: var(--app-secondary-color);
+      }
+
+      @media (max-width: 767px) {
+        :host {
+          margin: 0 12px;
+        }
+      }
+
+    </style>
+
+    <span id="ampContent"></span>
+
+  </template>
+
+  <script>
+
+    class AmpMyEmbed extends Polymer.Element {
+
+      static get is() { return 'amp-my-embed'; }
+
+      static get properties() { return {
+
+        url: {
+          type: String,
+          value: window.location.origin
+        },
+        fullUrl: {
+          type: String,
+          computed: '_computeFullUrl(url, itemUrl)',
+          observer: 'urlChanged'
+        },
+        itemUrl: {
+          type: String,
+          notify: true
+        },
+        content: {
+          type: String,
+          notify: true
+        }
+      };
+    }
+
+      _computeFullUrl(url, itemUrl) {
+        return [url, itemUrl, "embed"].join('')
+      }
+
+      urlChanged() {
+        this.fetchAndAttachAmpDoc_(this.fullUrl);
+      }
+
+      constructor() {
+        super();
+        // 'offline' is set to true if and when the document fetch fails.
+        this.state = {'offline': false};
+        this.ampReadyPromise_ = new Promise(resolve => {
+          (window.AMP = window.AMP || []).push(resolve);
+        });
+
+        this.xhr_ = null;
+        this._amp = null;
+        this._host = null;
+      }
+
+      ready() {
+        super.ready();
+        this.$.ampContent.textContent = this._amp;
+      }
+
+  /**
+   * Fetches the AMP document at `url` and attaches it as a shadow root.
+   * @private
+   * @param {string} url
+   */
+  fetchAndAttachAmpDoc_(url) {
+    this.fetchDocument_(url).then(doc => {
+      return this.ampReadyPromise_.then(amp => {
+
+        // Replace the old shadow root with a new div element.
+        const oldShadowRoot = this._amp;
+        this._host = document.createElement('div');
+        if (oldShadowRoot) {
+          this.$.ampContent.replaceChild(this._host, oldShadowRoot);
+        } else {
+          this.$.ampContent.appendChild(this._host);
+        }
+
+        this._amp = amp.attachShadowDoc(this._host, doc, url);
+      });
+    }).catch(error => {
+      this.setState({'offline': true});
+    });
+  }
+
+  /**
+   * Cleans up internal state of current shadow AMP document.
+   * @private
+   */
+  closeShadowAmpDoc_() {
+    if (typeof this._amp.close === 'function') {
+      this._amp.close();
+    }
+  }
+
+  /**
+   * Fetches and parses HTML at `url`.
+   * @private
+   * @param {string} url
+   * @return {!Promise<!Document|!string>} If fetch succeeds, resolved with {!Document}.
+   *         Otherwise, rejects with {!string} error description.
+   */
+  fetchDocument_(url) {
+    return new Promise((resolve, reject) => {
```

Comment:
Might be possible to use `fetch` here instead. Not sure what your browser matrix is, but there is a polyfill available.

Reply:
Added, thanks